core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerAggregator.java

```diff
@@ -0,0 +1,135 @@
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.batch.aggregator;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.etl.api.Emitter;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.BatchAggregator;
+import co.cask.cdap.etl.api.batch.BatchRuntimeContext;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Batch RowDenormalizer Aggregator Plugin - It is used to de-normalize data based on the key column.
+ */
+@Plugin(type = BatchAggregator.PLUGIN_TYPE)
+@Name("RowDenormalizer")
+@Description("This plugin is used to convert the raw data into de-normalized data based on the key column." +
+  "User is able to specify the list of fields that should be considered in denormalized record. " +
+  "Also, users have option to put alias for output field name. " +
+  "For example - " +
+  "‘ADDRESS’ in input is mapped to ‘addr’ in output schema. This denormalized data would be easy to query.")
+public class RowDenormalizerAggregator extends RecordAggregator {
+  private final RowDenormalizerConfig conf;
+  private Map<String, String> outputMappings;
+  private List<String> outputFields;
+  private Schema outputSchema;
+
+  public RowDenormalizerAggregator(RowDenormalizerConfig conf) {
+    super(conf.numPartitions);
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    validateInputFields(pipelineConfigurer.getStageConfigurer().getInputSchema());
+    pipelineConfigurer.getStageConfigurer().setOutputSchema(conf.parseSchema());
+  }
+
+  @Override
+  public void initialize(BatchRuntimeContext context) throws Exception {
+    outputFields = conf.getOutputFields();
+    outputMappings = conf.getOutputFieldMappings();
+  }
+
+  @Override
+  public void groupBy(StructuredRecord record, Emitter<StructuredRecord> emitter) throws Exception {
+    String keyField = conf.getKeyField();
+    Schema keyFieldSchema = Schema.recordOf("record", Schema.Field.of(keyField, Schema.of(Schema.Type.STRING)));
+    StructuredRecord.Builder builder = StructuredRecord.builder(keyFieldSchema);
+    builder.set(keyField, record.get(keyField));
+    emitter.emit(builder.build());
+  }
+
+  @Override
+  public void aggregate(StructuredRecord groupKey, Iterator<StructuredRecord> iterator,
+                        Emitter<StructuredRecord> emitter) throws Exception {
+    if (!iterator.hasNext()) {
+      return;
+    }
+    if (outputSchema == null) {
+      initializeOutputSchema();
+    }
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+    builder.set(conf.getKeyField(), groupKey.get(groupKey.getSchema().getFields().get(0).getName()));
```

Comment:
RowDenormalizerAggregator class extends RecordAggreagtor which further extends BatchAggregator class, so the overridden method requires the StructuredRecord as input.

If we need to change the method signature, then upper hierarchy needs to be changed.
Please let me know your inputs.